demo/demo/tests/test_apple_webhook.py

```diff
@@ -0,0 +1,264 @@
+        {'status': AppleValidationStatus.MALFORMED_DATA_OR_SERVICE_ISSUE.value}
+    )
+    with mock.patch(RECEIPT_FETCH_FUNCTION) as mock_receipt:
+        mock_receipt.return_value = receipt_data
+        with pytest.raises(AppleReceiptValidationError):
+            user_client.post(APPLE_API_WEBHOOK, make_receipt_query(), content_type='application/json')
+
+    assert len(SubscriptionPayment.objects.all()) == 0
+
+
+
+def test__app_store_notification__renew__product_id_changed(user_client,
+                                                            apple_bundle_id,
+                                                            apple_product_id):
+    transaction_id = 'special-transaction-id'
+    # Create the subscription via API.
+    receipt_data = make_receipt_data(
+        apple_product_id,
+        apple_bundle_id,
+        transaction_id=transaction_id,
+        original_transaction_id=transaction_id
+    )
+    with mock.patch(RECEIPT_FETCH_FUNCTION) as mock_receipt:
+        mock_receipt.return_value = receipt_data
+        response = user_client.post(APPLE_API_WEBHOOK, make_receipt_query(), content_type='application/json')
+        assert response.status_code == 200
+
+    # Provide a notification with a different product id.
+    notification_data = make_notification_data(
+        apple_bundle_id,
+        apple_product_id + 'x',
+        original_transaction_id=transaction_id
+    )
+    with mock.patch(NOTIFICATION_PARSER) as mock_parser:
+        mock_parser.return_value = notification_data
+        with pytest.raises(ProductIdChangedError):
+            user_client.post(APPLE_API_WEBHOOK, make_notification_query(), content_type='application/json')
+
+
+def test__app_store_notifications__renew__subscription_extended(user_client,
+                                                                apple_bundle_id,
+                                                                apple_product_id,
+                                                                user,
+                                                                apple_in_app):
+    transaction_id = 'special-transaction-id'
+    # Create the subscription via API.
+    receipt_data = make_receipt_data(
+        apple_product_id,
+        apple_bundle_id,
+        transaction_id=transaction_id,
+        original_transaction_id=transaction_id
+    )
+    with mock.patch(RECEIPT_FETCH_FUNCTION) as mock_receipt:
+        mock_receipt.return_value = receipt_data
+        response = user_client.post(APPLE_API_WEBHOOK, make_receipt_query(), content_type='application/json')
+        assert response.status_code == 200, response.content
+
+    # Receive a notification about subscription extension.
+    notification_data = make_notification_data(
+        apple_bundle_id,
+        apple_product_id,
+        original_transaction_id=transaction_id
+    )
+    with mock.patch(NOTIFICATION_PARSER) as mock_parser:
+        mock_parser.return_value = notification_data
```

Comment:
I think you can do it in one line, smth like `with mock.patch(..., return_value=notification_date):`